File: src/main/java/io/lsdconsulting/lsd/distributed/access/model/InterceptedInteraction.java
package io.lsdconsulting.lsd.distributed.access.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.ZonedDateTime;
import java.util.Collection;
import java.util.Map;

@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
public class InterceptedInteraction {
    String traceId;
    String body;
    Map<String, Collection<String>> requestHeaders;
    Map<String, Collection<String>> responseHeaders;
    String serviceName; // the calling service or the publisher or consumer
    String target; // the called service or the exchange name
    String path; // the called URL or the exchange name
    String httpStatus;
    String httpMethod;
    InteractionType interactionType;
    String profile;
    Long elapsedTime;
    ZonedDateTime createdAt;
}

File: src/main/kotlin/io/lsdconsulting/lsd/distributed/access/model/InterceptedInteraction.kt
package io.lsdconsulting.lsd.distributed.access.model

import java.time.ZonedDateTime

data class InterceptedInteraction(
    val traceId: String,
    val body: String? = null,
    val requestHeaders: MutableMap<String, Collection<String>> = mutableMapOf(),
    val responseHeaders: Map<String, Collection<String>> = mutableMapOf(),
    val serviceName: String, // the calling service or the publisher or consumer
    val target: String, // the called service or the exchange/topic name
    val path: String, // the called URL or the exchange/topic name
    val httpStatus: String? = null,
    val httpMethod: String? = null,
    val interactionType: InteractionType,
    val profile: String? = null,
    val elapsedTime: Long,
    val createdAt: ZonedDateTime,
)
